src/core/server/saved_objects/migrationsv2/integration_tests/type_registrations.test.ts

```diff
@@ -0,0 +1,136 @@
+
+// Types should NEVER be removed from this array
+const previouslyRegisteredTypes = [
+  'action',
+  'action_task_params',
+  'alert',
+  'api_key_pending_invalidation',
+  'apm-indices',
+  'apm-server-schema',
+  'apm-telemetry',
+  'app_search_telemetry',
+  'application_usage_daily',
+  'application_usage_totals',
+  'application_usage_transactional',
+  'background-session',
+  'canvas-element',
+  'canvas-workpad',
+  'canvas-workpad-template',
+  'cases',
+  'cases-comments',
+  'cases-configure',
+  'cases-connector-mappings',
+  'cases-sub-case',
+  'cases-user-actions',
+  'config',
+  'core-usage-stats',
+  'dashboard',
+  'endpoint:user-artifact',
+  'endpoint:user-artifact-manifest',
+  'enterprise_search_telemetry',
+  'epm-packages',
+  'epm-packages-assets',
+  'event_loop_delays_daily',
+  'exception-list',
+  'exception-list-agnostic',
+  'file-upload-telemetry',
+  'file-upload-usage-collection-telemetry',
+  'fleet-agent-actions',
+  'fleet-agent-events',
+  'fleet-agents',
+  'fleet-enrollment-api-keys',
+  'fleet-preconfiguration-deletion-record',
+  'graph-workspace',
+  'index-pattern',
+  'infrastructure-ui-source',
+  'ingest-agent-policies',
+  'ingest-outputs',
+  'ingest-package-policies',
+  'ingest_manager_settings',
+  'inventory-view',
+  'kql-telemetry',
+  'legacy-url-alias',
+  'lens',
+  'lens-ui-telemetry',
+  'map',
+  'maps-telemetry',
+  'metrics-explorer-view',
+  'ml-job',
+  'ml-module',
+  'ml-telemetry',
+  'monitoring-telemetry',
+  'osquery-saved-query',
+  'osquery-usage-metric',
+  'query',
+  'sample-data-telemetry',
+  'search',
+  'search-session',
+  'search-telemetry',
+  'security-rule',
+  'security-solution-signals-migration',
+  'siem-detection-engine-rule-actions',
+  'siem-detection-engine-rule-status',
+  'siem-ui-timeline',
+  'siem-ui-timeline-note',
+  'siem-ui-timeline-pinned-event',
+  'space',
+  'spaces-usage-stats',
+  'tag',
+  'task',
+  'telemetry',
+  'timelion-sheet',
+  'tsvb-validation-telemetry',
+  'ui-counter',
+  'ui-metric',
+  'upgrade-assistant-ml-upgrade-operation',
+  'upgrade-assistant-reindex-operation',
+  'upgrade-assistant-telemetry',
+  'uptime-dynamic-settings',
+  'url',
+  'usage-counters',
+  'visualization',
+  'workplace_search_telemetry',
+].sort();
+
+describe('SO type registrations', () => {
+  it('does not remove types from registrations without updating unusedTypesQuery', async () => {
+    const root = kbnTestServer.createRoot({}, { oss: false });
+    const setup = await root.setup();
+    const currentlyRegisteredTypes = setup.savedObjects
+      .getTypeRegistry()
+      .getAllTypes()
+      .map(({ name }) => name)
+      .sort();
+    await root.shutdown();
+
+    // Make sure that all `REMOVED_TYPES` are in `previouslyRegisteredTypes`
+    expect(previouslyRegisteredTypes.filter((type) => REMOVED_TYPES.includes(type))).toEqual(
+      REMOVED_TYPES // Use array comparison for readable test failure messages
+    );
```

Comment:
What is this specific test exactly asserting?

Reply:
This validates that if someone adds a type to `REMOVED_TYPES` it also gets added to `previouslyRegisteredTypes`. This prevents the case where a type gets removed from `REMOVED_TYPES` by accident.